src/software/embedded/robot_localizer.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <Eigen/Dense>
+#include <deque>
+#include "software/geom/angle.h"
+#include "software/geom/angular_velocity.h"
+#include "software/sensor_fusion/filter/kalman_filter.h"
+#include "proto/world.pb.h"
+#include "time.h"
+#include "software/embedded/services/imu.h"
+
+
+class RobotLocalizer {
+public:
+    RobotLocalizer(double process_noise_variance, double vision_noise_variance, double encoder_noise_variance, double target_angular_acceleration_variance):
+    filter_(),
+    process_noise_variance_(process_noise_variance),// assumes process noise is a discrete time wiener process
+    history()
+    {
+        filter_.P << 30, 0,  0,
+                     0,  4, 0,
+                     0,  0,  5;
+        filter_.B << 0.0, 1, 0.0;
+        filter_.R << vision_noise_variance, 0.0,                    0.0,0.0,
+                     0.0,                   encoder_noise_variance, 0.0,0.0,
+                     0.0,                   0.0,                    ImuService::IMU_VARIANCE,0.0,
+                     0.0,                   0.0,                    0.0,target_angular_acceleration_variance;
+        filter_.x = Eigen::Matrix<double, 3, 1>(0.0, 0.0, 0.0);
+        clock_gettime(CLOCK_MONOTONIC, &current_time_);
+        clock_gettime(CLOCK_MONOTONIC, &last_step_);
+
+    }
+
+    void Step(AngularVelocity targetAcceleration);
+
+    /**
+     *
+     * @param orientation Vision reading of the orientation of the robot in world space
+     * @param ageSeconds Age in seconds of the vision snapshot (time since it was taken)
+     */
+    void RollbackVision(Angle orientation, double ageSeconds);
+
+    /**
+     *
+     * @param orientation Vision reading of the orientation of the robot in world space
+     */
+    void UpdateVision(Angle orientation);
```

Comment:
```suggestion
    void updateVision(const Angle& orientation);
```
functions use [camelCase](https://github.com/UBC-Thunderbots/Software/blob/master/docs/code-style-guide.md#names-and-variables)